modules/bash-commons/src/assert.sh

```diff
@@ -70,6 +70,34 @@ function assert_value_in_list {
   fi
 }
 
+# Reads in a list of keys and values and asserts that one and only one of the values is set.
+# This is useful for command line options that are mutually exclusive.
+# Example:
+#   assert_exactly_one_of "--opt1" "" "--opt2" "val2" "--opt3" "" "--opt4" ""
+# Examples that assert failure:
+#   assert_exactly_one_of "--opt1" "val1" "--opt2" "val2" "--opt3" "" "--opt4" ""
+#   assert_exactly_one_of "--opt1" "" "--opt2" "" "--opt3" "" "--opt4" ""
+function assert_exactly_one_of {
+  local -ra args=("$@")
+  local -r num_args="${#args[@]}"
+
+  local num_non_empty=0
+  local -a arg_names=()
+
+  # Determine how many arg_vals are non-empty
+  for (( i=0; i<$((num_args+1)); i+=2 )); do
```

Comment:
Might want to do a check that the length is divisble by 2 first

Reply:
I was thinking about that, but if you dereference a bash array element that isn't set, it just returns `""` which wouldn't trigger the next line and would be okay. Happy to change it if you prefer though.

Reply:
I think it would be better to show a clear error here, as in most cases, passing an add number of arguments will be an accidental bug. Thx!

Reply:
Added the check and a test